### ISSUE TITLE
chromedriver Issues (MacOS Mojave)

### ISSUE THREAD

**Issue by MTPeyton** [InstaPy/InstaPy#3488]
## Expected Behavior
quickstart.py should run as expected. Instead it terminates with an error about chromedriver.

## Current Behavior
```
ERROR [2018-12-03 16:11:18] [umiresearch]  Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home
Traceback (most recent call last):
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 76, in start
    stdin=PIPE)
  File "/Users/mpeyton/.pyenv/versions/3.7.0/lib/python3.7/subprocess.py", line 756, in __init__
    restore_signals, start_new_session)
  File "/Users/mpeyton/.pyenv/versions/3.7.0/lib/python3.7/subprocess.py", line 1499, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver': '/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/instapy/instapy.py", line 357, in set_selenium_local_session
    chrome_options=chrome_options)
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/chrome/webdriver.py", line 73, in __init__
    self.service.start()
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 83, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home

ERROR:umiresearch:Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home
Traceback (most recent call last):
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 76, in start
    stdin=PIPE)
  File "/Users/mpeyton/.pyenv/versions/3.7.0/lib/python3.7/subprocess.py", line 756, in __init__
    restore_signals, start_new_session)
  File "/Users/mpeyton/.pyenv/versions/3.7.0/lib/python3.7/subprocess.py", line 1499, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver': '/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/instapy/instapy.py", line 357, in set_selenium_local_session
    chrome_options=chrome_options)
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/chrome/webdriver.py", line 73, in __init__
    self.service.start()
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 83, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home

Traceback (most recent call last):
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 76, in start
    stdin=PIPE)
  File "/Users/mpeyton/.pyenv/versions/3.7.0/lib/python3.7/subprocess.py", line 756, in __init__
    restore_signals, start_new_session)
  File "/Users/mpeyton/.pyenv/versions/3.7.0/lib/python3.7/subprocess.py", line 1499, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver': '/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/instapy/instapy.py", line 357, in set_selenium_local_session
    chrome_options=chrome_options)
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/chrome/webdriver.py", line 73, in __init__
    self.service.start()
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/selenium/webdriver/common/service.py", line 83, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "instagram_invite_responder.py", line 15, in <module>
    headless_browser=True)
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/instapy/instapy.py", line 232, in __init__
    self.set_selenium_local_session()
  File "/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/instapy/instapy.py", line 361, in set_selenium_local_session
    Settings.chromedriver_location))
instapy.instapy.InstaPyError: ensure chromedriver is installed at /Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/assets/chromedriver
```

## Possible Solution (optional)
I already have Google Chrome installed on my machine (Chrome 70). I have tried installing chromedriver several ways.

I first tried using `pip install chromedriver` in my virtualenv Python environment that I made for this project. That ended up installing chromedriver at `/Users/mpeyton/.pyenv/versions/umienv/lib/python3.7/site-packages/chromedriver` NOT in the assets folder. I tried making an assets folder and moving chromedriver to that. I then got new error messages stating that it had incorrect permissions. I ran 'chmod 755 assets' and 'chmod 755 ./assets/chromedriver'. Then ran quickstart again. The same permissions errors occurred.

I then uninstalled chromedriver using `pip uninstall chromedriver`. Then i used `brew cask install chromedriver`. This installed chromedriver to my /usr/bin folder (which is already in my PATH). But I got the same errors as before (chromedriver not in PATH).

Finally I uninstalled that using `brew cask uninstall chromedriver` and then manually downloaded chromedriver from its website. I placed it in a folder, then added that folder to PATH. Still did not work.

Final note: I tested each of these solutions with all possible combinations of `nogui` and `headless_browser` options.

## InstaPy Configuration

```
from instapy import InstaPy
from instapy.util import smart_run

insta_username = 
insta_password = 

session = InstaPy(username=insta_username,
                  password=insta_password,
                  headless_browser=True)

with smart_run(session):

    session.login()

    uber_followers = session.grab_followers(username="uber", amount=5, live_match=False, store_locally=False)
    print(uber_followers)

    session.end()
```

**Comment by visualheroes** [comment]
Hey i'm running Instapy on OSX Mojave and High Sierra without a problem.
You don't need to install chromedriver as a package, only in the asset folder (no chmod needed).
How did you set up your virtual env?  Did you use Python 2.7?

Basically you just need to install Xcode tools then Homebrew  followed by Python 3. After that just create your virtualenv and add all the required packages (requirements file).

**Comment by MTPeyton** [comment]
Hi visualheroes, thanks for the response. I think I just misunderstood how the project directory needed to be structured. I have it working now. I think I also need to read up a bit more on how virtualenv works, because I think that confused me a lot.

*In case anyone is wondering what I did*

I was trying to create all of this in my own, new private GitHub repo. I ended up deleting everything in my repo, then I cloned the InstaPy repo to a new folder. I copied everything from that folder to the folder of my private repo (manually, because apparently --mirror was the wrong thing to try lol). I then pushed up to my private repo. I then went into the assets folder in my private repo and added the chromedriver executable which I manually downloaded from the website. Finally I ran `pip install .` Now I can run quickstart.py.

Note: I couldn't fork the InstaPy repo because my repo was already made and owned by an organization.
